### PR TITLE
libpng 1.6.38: pass PNG_EXECUTABLES=OFF + fix v2 linter error

### PR DESCRIPTION
**Pull request by kambala-decapitator**
Specify library name and version:  **libpng/1.6.38**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

v1.6.38 introduced new CMake variable `PNG_EXECUTABLES`, now building executables can be disabled as they're removed from package anyway.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libpng/all/conanfile.py [hunk 1]
```diff
@@ -37,6 +38,7 @@ class LibpngConan(ConanFile):
         "sse": True,
         "vsx": True,
         "api_prefix": "",
+        "executables": True,
```
Comment (SpaceIm) [inline recipes/libpng/all/conanfile.py:41]: this option doesn't really make sense since these utilities are removed in package() anyway.

Comment (kambala-decapitator) [reply recipes/libpng/all/conanfile.py:41]: should `PNG_EXECUTABLES=OFF` be set unconditionally then?

Comment (SpaceIm) [reply recipes/libpng/all/conanfile.py:41]: yes

Comment (kambala-decapitator) [reply recipes/libpng/all/conanfile.py]: thanks, adjusted

**Comment by kambala-decapitator** [comment]
should I wait until v1.53 is installed on CI?

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**